src/AppContainer.jsx

```diff
@@ -71,102 +71,73 @@ function AppContainer() {
     setState(state => ({...state, data: locale === 'pl' ? dataPL : dataEN}));
   }, [locale]);
 
-  const renderStreetListWithMap = props => (
+  const renderLayout = (children: Node) => (
     <>
-      <Navigation streetName="" />
-      <MapContainer data={data} {...props} />
-      <StreetList mapView data={data} {...props} />
-    </>
-  );
-
-  const renderStreetList = props => (
-    <>
-      <Navigation streetName="" />
-      <StreetList data={data} {...props} />
-    </>
-  );
-
-  const renderDetail = () => <StreetDetail data={data} />;
-
-  const renderDesktopView = props => (
-    <>
-      <NavigationDesktop />
-      <div className={style.desktopViewContainer}>
-        <StreetList desktopView data={data} {...props} />
-        <MapContainer data={data} {...props} />
+      <Navigation data={data} />
+      <div className={style.container}>
+        {children}
       </div>
     </>
   );
 
-  const renderDetailDesktop = props => (
-    <>
-      <NavigationDesktop />
-      <div className={style.desktopViewContainer}>
-        <StreetDetail desktopView data={data} />
-        <MapContainer data={data} {...props} />
-      </div>
-    </>
+  // todo: pass selected item to props, this should prevent rendering whole component tree
```

Comment:
the idea here is to have single layout tree and components in content with `selectedItem` already given to them (if they base on that (like detail or map)

This would prevent re-rendering of all components on the way